File: validation/poc-00/java-candidate/src/main/java/dev/vra/poc00/infrastructure/MigrateLocal.java
package dev.vra.poc00.infrastructure;

import org.flywaydb.core.Flyway;

/** Separate process; the running backend never invokes this command. */
public final class MigrateLocal {
    private MigrateLocal() {}
    public static void main(String[] args) {
        var settings = LocalDatabaseSettings.fromEnvironment();
        Flyway.configure().dataSource(settings.url(), settings.username(), settings.password())
                .locations("classpath:db/migration").cleanDisabled(true)
                .load().migrate();
    }
}

File: validation/poc-00/kotlin-candidate/src/main/kotlin/dev/vra/poc00/kotlin/infrastructure/MigrateLocal.kt
package dev.vra.poc00.kotlin.infrastructure

import org.flywaydb.core.Flyway

/** Explicit migration process; the web application never runs this automatically. */
object MigrateLocal {
    @JvmStatic
    fun main(args: Array<String>) {
        val settings = LocalDatabaseSettings.fromEnvironment()
        Flyway.configure()
            .dataSource(settings.url, settings.username, settings.password)
            .locations("classpath:db/migration")
            .cleanDisabled(true)
            .load()
            .migrate()
    }
}
